Change Guzzle HTTP Adapter to PSR Adapter

Firesphere opened the pull request:
# Hi there and thank you for taking the time and effort to make a Pull Request

No, seriously, we are delighted you made time and effort to help us, even in the smallest way!


# Please check the following

- [x] Have you opened an issue to discuss the feature and agree its general design?
- [x] Do you have a use case and, ideally, an example program using the feature?
- [x] Do you have tests covering 90%+ of the feature code (and, of course passing)
- [x] Have you written complete and accurate doc comments?
- [x] Have you updated the README or docs where needed?
- [x] Code is readable
- [x] Code is tested
- [x] I've read the Code of Conduct (Or, at least, I do believe I agree with it)
- [x] Frontend features don't look completely uncared for
- [x] Code is PSR-2 compliant
- [x] I'm fairly sure this is useful
- [x] I'm not just making this PR for HacktoberFest
- [x] You rock. Thanks a lot.

Commit:
Change Guzzle HTTP Adapter to PSR Adapter

require at least PHP7.3

Solr 9 compatibility

Flatten hierarchy to avoid Solr thinking it's a child document

Add required PSR adapters

Firesphere commented:
#247

You should be able to use `dev-hans/guzzle-to-psr` for PHP8 compatibility.

No promises though!

Firesphere commented:
@marczhermo Could you please have a look at your Docker container, to update the PHP version to 7.3+, and maybe/possibly PHP8 as a secondary build track?

marczhermo commented on src/Services/SolrCoreService.php:
```diff
@@ -17,6 +17,8 @@
 use Firesphere\SolrSearch\Traits\CoreServiceTrait;
 use GuzzleHttp\Client as GuzzleClient;
 use GuzzleHttp\HandlerStack;
+use Http\Discovery\HttpClientDiscovery;
```
Hi @Firesphere, 
Can we add `php-http/discovery:^1.14` to `composer.json`?
I got this error below:
```
[Emergency] Uncaught Error: Class 'Http\Discovery\HttpClientDiscovery' not found
```

Firesphere replied:
Yes, Of course!

Firesphere commented:
Hey Marco,

I've added and updated this PR, it's now ready to go. Test failing is unrelated and work locally, I'll fix up the tests on master when I get to it.

This adds:
- Solarium 6
- Solr 9
- PSR18 instead of Guzzle

ssmarco reviewed (approved):
Solr 9 :tada:

Firesphere committed:
Merge branch 'primary' into hans/guzzle-to-psr
